src/web/app/pages-maintainer/maintainer-page.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { environment } from '../../environments/environment';
+import { AuthService } from '../../services/auth.service';
+import { AuthInfo } from '../../types/api-output';
+
+/**
+ * Base skeleton for maintainer pages.
+ */
+@Component({
+  selector: 'tm-maintainer-page',
+  templateUrl: './maintainer-page.component.html',
+  styleUrls: ['./maintainer-page.component.scss'],
+})
+export class MaintainerPageComponent implements OnInit {
+
+  user: string = '';
+  institute?: string = '';
+  isInstructor: boolean = false;
+  isStudent: boolean = false;
+  isAdmin: boolean = false;
+  isMaintainer: boolean = false;
+  navItems: any[] = [
+    {
+      url: '/web/maintainer',
+      display: 'Home',
+    },
+  ];
+  isFetchingAuthDetails: boolean = false;
+
+  private backendUrl: string = environment.backendUrl;
+
+  constructor(private route: ActivatedRoute, private authService: AuthService) {}
+
+  ngOnInit(): void {
+    this.isFetchingAuthDetails = true;
+    this.route.queryParams.subscribe((queryParams: any) => {
+      this.authService.getAuthUser(queryParams.user).subscribe((res: AuthInfo) => {
+        if (res.user) {
+          this.user = res.user.id;
+          if (res.masquerade) {
+            this.user += ' (M)';
+          }
+          this.institute = res.institute;
+          this.isInstructor = res.user.isInstructor;
+          this.isStudent = res.user.isStudent;
+          this.isAdmin = res.user.isAdmin;
+          this.isMaintainer = res.user.isMaintainer;
+        } else {
+          window.location.href = `${this.backendUrl}${res.instructorLoginUrl}`;
```

Comment:
Why is this redirected to instructor's login URL?

Reply:
I missed that part out. maintainer's login URL is now added.